### DISCUSSION TITLE
I don't know why I can't connect

### DISCUSSION THREAD

**Discussion by qq921124136** [discord-jda/JDA#2517] [Questions and Help] [answered]
![123](https://github.com/discord-jda/JDA/assets/83702287/153aba62-c2d7-4e37-a920-077bd59bd75e)
![image](https://github.com/discord-jda/JDA/assets/83702287/6f8e33c7-9275-4e35-b4b9-9b08c015d990)
![image](https://github.com/discord-jda/JDA/assets/83702287/3da9690f-bb44-43b0-9a7c-d7b71e2261fb)
![image](https://github.com/discord-jda/JDA/assets/83702287/a18c9afc-9332-4e88-bb25-699d588185a2)

**Comment by MinnDevelopment** [comment, marked as answer]
You haven't configured the proxy for the websocket client. I believe this would be how:

```java
WebSocketFactory factory = new WebSocketFactory();
factory.getProxySettings()
  .setHost("127.0.0.1")
  .setPort(7890);

builder.setWebsocketFactory(factory)
```

See:
- [JDABuilder#setWebsocketFactory](https://docs.jda.wiki/net/dv8tion/jda/api/JDABuilder.html#setWebsocketFactory(com.neovisionaries.ws.client.WebSocketFactory))
- [WebSocketFactory](https://takahikokawasaki.github.io/nv-websocket-client/com/neovisionaries/ws/client/WebSocketFactory.html)
- [ProxySettings](https://takahikokawasaki.github.io/nv-websocket-client/com/neovisionaries/ws/client/ProxySettings.html)

**Reply by qq921124136** [reply]
![image](https://github.com/discord-jda/JDA/assets/83702287/6c833f46-1b40-4b5b-94c0-8980349f80a5)
**谢谢**
**Thank you very much**
